### PR TITLE
Update sdk version to 0.4.2, add the `alloc` example

### PR DESCRIPTION
**Pull request by jayz22**
### What

Update sdk version to 0.4.2, add the `alloc` example

### Why

[TODO: Why this change is being made. Include any context required to understand the why.]

### Known limitations

[TODO or N/A]

<a href="https://gitpod.io/#https://github.com/stellar/soroban-examples/pull/193"></a>

### CODE REVIEW

File: alloc/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+license = "Apache-2.0"
+edition = "2021"
+publish = false
+
+[lib]
+crate-type = ["cdylib"]
+doctest = false
+
+[dependencies]
+soroban-sdk = { workspace = true }
```
Comment (leighmcculloch) [inline alloc/Cargo.toml:14]:
```suggestion
soroban-sdk = { workspace = true, features = ["alloc"] }
```

Comment (jayz22) [reply alloc/Cargo.toml:14]: Ah, thanks that fixes the problem. Not sure why I only added it to the dev dependencies.

**Comment by leighmcculloch** [comment]
Looks great! Thanks @jayz22!

Are you also planning on adding a page about the example to the docs at https://soroban.stellar.org/docs/category/examples? We should add a page at the bottom of the example list I think where we discuss the allocator provided.

**Comment by jayz22** [comment]
> Looks great! Thanks @jayz22!
> 
> Are you also planning on adding a page about the example to the docs at https://soroban.stellar.org/docs/category/examples? We should add a page at the bottom of the example list I think where we discuss the allocator provided.

Yes I will do that.

**Comment by jayz22** [comment]
@leighmcculloch I've added the corresponding docs example in here: https://github.com/stellar/soroban-docs/pull/265

**Pull request merged**